pkg/controllers/allocation/controller.go

```diff
@@ -23,66 +23,80 @@ import (
 	"github.com/awslabs/karpenter/pkg/cloudprovider"
 	"github.com/awslabs/karpenter/pkg/packing"
 	"github.com/awslabs/karpenter/pkg/utils/apiobject"
+	"golang.org/x/time/rate"
 
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/util/workqueue"
+	controllerruntime "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 // Controller for the resource
 type Controller struct {
 	filter        *Filter
 	binder        *Binder
+	batcher       *Batcher
 	constraints   *Constraints
 	packer        packing.Packer
 	cloudProvider cloudprovider.CloudProvider
-}
-
-// For returns the resource this controller is for.
-func (c *Controller) For() client.Object {
-	return &v1alpha2.Provisioner{}
-}
-
-func (c *Controller) Interval() time.Duration {
-	return 5 * time.Second
-}
-
-func (c *Controller) Name() string {
-	return "provisioner/allocator"
+	kubeClient    client.Client
 }
 
 // NewController constructs a controller instance
-func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface, cloudProvider cloudprovider.CloudProvider) *Controller {
+func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface, cloudProvider cloudprovider.CloudProvider,
+	MaxBatchWindow time.Duration, BatchIdleTimeout time.Duration) *Controller {
```

Comment:
nit, function parameters shouldn't use capital letters `MaxBatchWindow`, but IMO you should keep them capital and just make them file constants.